packages/vite/src/node/plugins/html.ts

```diff
@@ -518,9 +529,21 @@ export function buildHtmlPlugin(config: ResolvedConfig): Plugin {
                       })
                       js += importExpression
                     } else {
+                      // If the node is a link, check if it can be inlined. If not, set `shouldInline`
+                      // to `false` to force no inline. If `undefined`, it leaves to the default heuristics.
+                      const isNoInlineLink =
+                        node.nodeName === 'link' &&
+                        node.attrs.some(
+                          (p) =>
+                            p.name === 'rel' && noInlineLinkRels.has(p.value),
```

Comment:
```suggestion
                            p.name === 'rel' && p.value.split(/[\t\n\f\r ]/).some(v => noInlineLinkRels.has(v.toLowerCase())),
```
[`rel` is a space-separated list](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/link#rel) and is case-insensitive. I guess it's better to check all values. 

spec: [about space-separated list](https://html.spec.whatwg.org/multipage/links.html#linkTypes:~:text=To%20determine%20which%20link%20types%20apply%20to%20a%20link%2C%20a%2C%20area%2C%20or%20form%20element%2C%20the%20element%27s%20rel%20attribute%20must%20be%20split%20on%20ASCII%20whitespace.%20The%20resulting%20tokens%20are%20the%20keywords%20for%20the%20link%20types%20that%20apply%20to%20that%20element.), [about case-insensitivity](https://html.spec.whatwg.org/multipage/links.html#linkTypes:~:text=Keywords%20are%20always%20ASCII%20case%2Dinsensitive%2C%20and%20must%20be%20compared%20as%20such.)

Reply:
From my experience these assets name tends to be correctly written an unique, Apple probably doesn't do that much effort to display icons & manifest so personally I'm fine with the current check